postgres/__init__.py

```diff
@@ -838,6 +840,30 @@ def make(self, values):
             instance = ModelSubclass(record)
             return instance
 
+        def parse(self, s, curs, retry=False):
+            # Override to protect against race conditions:
+            #   https://github.com/gratipay/postgres.py/issues/26
+
+            if s is None:
+                return None
+
+            tokens = self.tokenize(s)
+            if len(tokens) != len(self.atttypes):
+                # The number of columns has changed, re-fetch the type info
+                self.__dict__.update(self._from_db(self.name, curs).__dict__)
```

Comment:
Where did we get this line from, and why do we not call it in the normal course of operation? Why are we doing something in this case that we don't otherwise do? Don't we need to configure the class when it's first instantiated? Why am I not seeing that in our code? Is it in `psycopg2`?